rolling-shutter/snapshot/handler.go

```diff
@@ -117,31 +111,31 @@ func (handler *EonPublicKeyHandler) ValidateMessage(_ context.Context, msg p2pms
 
 func (handler *DecryptionKeyHandler) HandleMessage(ctx context.Context, m p2pmsg.Message) ([]p2pmsg.Message, error) {
 	var result []p2pmsg.Message
-	key := m.(*p2pmsg.DecryptionKey)
+	keys := m.(*p2pmsg.DecryptionKeys)
 	db := database.New(handler.dbpool)
 
-	rows, err := db.InsertDecryptionKey(
-		ctx, database.InsertDecryptionKeyParams{
-			EpochID: key.EpochID,
-			Key:     key.Key,
-		},
-	)
-	if err != nil {
-		return result, err
-	}
-
-	// already seen
-	if rows == 0 {
-		return result, nil
+	for _, key := range keys.Keys {
+		rows, err := db.InsertDecryptionKey(
```

Comment:
here we potentially should first insert all keys and then submit them to the api?
I'm not sure how the recovery case is handled, but I think we don't have retries currently.
So like this at least the keys itself are not lost if there is an arbitrary error in the api.